awx/api/views/inventory.py

```diff
@@ -115,6 +115,15 @@ class InventoryInputInventoriesList(SubListAttachDetachAPIView):
     parent_model = Inventory
     relationship = 'input_inventories'
 
+    # Specifically overriding the post method on this view to disallow constructed inventories as input inventories
+    def post(self, request, *args, **kwargs):
+        obj = Inventory.objects.get(id=request.data.get('id'))
```

Comment:
It would be good to implement this at some lower level, so that we do not re-invent the wheel like with this logic to get the object.

Reply:
this method was made for this purpose:

https://github.com/ansible/awx/blob/0815f935ca359537c023efadf286671e31dc3374/awx/api/generics.py#L633-L635

So here, I think `sub` is the input inventory, so you can check `obj.kind` without having to do `Inventory.objects.get`, which would obligate you to do error handling if you couldn't get the object.